### PR TITLE
Warn if one of the three redis nodes are down

### PR DESCRIPTION
**Pull request by meikomeik**
Change redis cluster status check to only warn if one of the three nodes are down

### CODE REVIEW

File: src/check_redis_cluster.py [hunk 1]
```diff
@@ -35,28 +32,23 @@
...
-    slave_addr, slave_state, cluster_state_slave = _get_cluster_status(
+    slave_addr, slave_state, cluster_state_slave, failed_slave = _get_cluster_status(
         args.slave_port, args.password)
-
     if master_state != 'unknown' and slave_state != 'unknown':
         if cluster_state_master != 'ok' and cluster_state_slave != 'ok':
             print('CRITICAL - cluster is broken')
             code = 2
-        elif master_state != 'master' or slave_state != 'slave':
+        elif len(failed_master + failed_slave) > 0:
```
Comment (TheFRedFox) [inline src/check_redis_cluster.py:48]:
```suggestion
        failed_hosts = failed_master + failed_slave
        elif failed_hosts:
```

File: src/check_redis_cluster.py [hunk 2]
```diff
@@ -35,28 +32,23 @@
...
             print('WARNING - cluster status is degraded')
-            if master_state != 'master':
-                print('{} got demoted to slave'.format(master_addr))
-                code = 1
-            if slave_state != 'slave':
-                print('{} got promoted to master'.format(slave_addr))
+            for host in failed_master + failed_slave:
```
Comment (TheFRedFox) [inline src/check_redis_cluster.py:50]:
```suggestion
            for host in failed_hosts:
```

**Review by lacunoc — APPROVED** [review]
looks good to me

File: src/check_redis_cluster.py (hunk 1 above)
Comment (meikomeik) [reply src/check_redis_cluster.py:48]: I added the requests changes (and further improved a few pep8 errors)

File: src/check_redis_cluster.py (hunk 2 above)
Comment (meikomeik) [reply src/check_redis_cluster.py:50]: I added the requests changes (and further improved a few pep8 errors)

**Pull request merged**